python/ml4ir/base/features/feature_config.py

```diff
@@ -116,32 +116,27 @@ def extract_features(self):
         - label
         - all_features
         """
-        try:
-            self.query_key = self.features_dict.get(FeatureConfigKey.QUERY_KEY)
-            self.all_features.append(self.query_key)
-        except KeyError:
-            self.query_key = None
+        for mandatory_key in [FeatureConfigKey.LABEL, FeatureConfigKey.FEATURES]:
```

Comment:
from the code I see that the `mandatory` should be 3 ( I count 3 `keyError` exceptions): 
`FeatureConfigKey.LABEL, FeatureConfigKey.FEATURES, FeatureConfigKey.QUERY_KEY` no?

Reply:
In the previous code, for QUERY_KEY, we catched a KeyError, but did not raise a KeyError.
In the new code, we check 'not in self.features_dict' 3 times too (2 for mandatory keys, one for QUERY_KEY line 124).

Reply:
There is also `RANK` but only for SequenceExample